api/v1/model_handlers.go

```diff
@@ -474,6 +490,76 @@ func (a *Api) podMetrics(request *restful.Request, response *restful.Response) {
 	response.WriteEntity(exportTimeseries(timeseries, new_stamp))
 }
 
+type AggregatedMetricPoints []AggregatedMetricPoint
+
+func (s AggregatedMetricPoints) Len() int {
+	return len(s)
+}
+func (s AggregatedMetricPoints) Swap(i, j int) {
+	s[i], s[j] = s[j], s[i]
+}
+func (s AggregatedMetricPoints) Less(i, j int) bool {
+	// we want the descending order
+	return s[j].Timestamp.Before(s[i].Timestamp)
+}
+
+// podMetrics returns a metric timeseries for a metric of the Pod entity.
+func (a *Api) podMetricsSum(request *restful.Request, response *restful.Response) {
```

Comment:
This is a much more "fat" API handler than the rest of the model handlers, which does not consider model locking. Due to the consecutive calls to GetPodMetric, it is possible that a model Update operation may be performed between these calls, leading to potential concurrency issues. 
For example, if "end" is specified as zero, it is possible that some of the pods will return a []TimePoint with some timestamps further in the future than the result of other pods.

To avoid this issue, I would recommend following the pattern used in the other handlers, where a single model method is called by the handler. This would involve the creation of the following model method:

GetPodListMetric(PodListRequest) []store.TimePoint

type PodListRequest struct {
NamespaceName string
PodNames []string
MetricName: string
Start: time.Time
End: time.Time
}